### PR TITLE
[CALCITE-5611] Show sql for failed tests for SqlOperatorTest

### PR DESCRIPTION
**Pull request by snuyanzin**
The PR  is aiming to make failing test showing SQL for which the test is failed

### CODE REVIEW

File: testkit/src/main/java/org/apache/calcite/sql/test/ResultCheckers.java [hunk 1]
```diff
@@ -103,11 +103,12 @@ public static SqlTester.ResultChecker isNullValue() {
    * @param refSet    Expected results
```
[the remark this answers is not in the document]
Comment (snuyanzin) [reply testkit/src/main/java/org/apache/calcite/sql/test/ResultCheckers.java:104]: thanks, done

**Comment by JiajunBernoulli** [comment]
Although this improvement is really helpful, but you changed many public interface.

I worry about they maybe cause damage to other users.

@snuyanzin Can you add interfaces instead of changing them?

**Comment by snuyanzin** [comment]
hmm.. there are only 2 interfaces from `testkit` changed `org.apache.calcite.sql.test.SqlTester.TypeChecker` and `org.apache.calcite.sql.test.SqlTester.ResultChecker`
and both about matcher checkers....
Even if so the migration is pretty straightforward by adding `String` field

**Comment by rubenada** [comment]
I agree with @snuyanzin , as far as I can see only two interfaces from testkit are modified, IMO that seems no big deal.

**Review by JiajunBernoulli — APPROVED** [review]

**Comment by rubenada** [comment]
LGTM. Needs to squash commits into a single one.

**Comment by snuyanzin** [comment]
squashed commits and rebased

**Pull request closed without merging**